camel-kafka-connector-catalog/src/main/java/org/apache/camel/kafkaconnector/catalog/CamelKafkaConnectorCatalog.java

```diff
@@ -0,0 +1,138 @@
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.camel.kafkaconnector.model.CamelKafkaConnectorModel;
+import org.apache.camel.kafkaconnector.model.CamelKafkaConnectorOptionModel;
+import org.apache.camel.tooling.model.JsonMapper;
+import org.apache.camel.util.json.JsonObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CamelKafkaConnectorCatalog {
+
+    static List<String> connectorsName = new ArrayList<String>();
+    static Map<String, CamelKafkaConnectorModel> connectorsModel = new HashMap<String, CamelKafkaConnectorModel>();
+    private static final Logger LOG = LoggerFactory.getLogger(CamelKafkaConnectorCatalog.class);
+    private static final String CONNECTORS_DIR = "src/generated/resources/connectors";
+    private static final String DESCRIPTORS_DIR = "src/generated/resources/descriptors";
+    private static final String CONNECTORS_PROPERTIES = "connectors.properties";
+
+    public CamelKafkaConnectorCatalog() {
+        initCatalog();
+        generateModel();
+    }
+
+    private void generateModel() {
+        for (String connector : connectorsName) {
+            connectorsModel.put(connector, getConnectorModel(connector));
+        }
+    }
+
+    private void initCatalog() {
+        try (FileInputStream input = new FileInputStream(DESCRIPTORS_DIR + File.separator + CONNECTORS_PROPERTIES)) {
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(input));
+
+            while (reader.ready()) {
+                String connector = reader.readLine();
+                if (connector.equalsIgnoreCase("camel-coap-tcp-source")) {
+                    connectorsName.add("camel-coap+tcp-source");
+                } else if (connector.equalsIgnoreCase("camel-coaps-tcp-source")) {
+                    connectorsName.add("camel-coaps+tcp-source");
+                } else if (connector.equalsIgnoreCase("camel-coaps-tcp-sink")) {
+                    connectorsName.add("camel-coaps+tcp-sink");
+                } else if (connector.equalsIgnoreCase("camel-coap-tcp-sink")) {
+                    connectorsName.add("camel-coap+tcp-sink");
+                } else {
+                    connectorsName.add(connector);
+                }
+            }
+        } catch (FileNotFoundException e) {
+            LOG.error("Cannot find file: {}", e.getMessage(), e);
+        } catch (IOException e) {
+            LOG.error("IO Exception: {}", e.getMessage(), e);
+        }
+    }
+
+    public String getConnectorAsJson(String connectorName) {
+        String result = null;
+        try {
+            result = Files.lines(Paths.get(CONNECTORS_DIR + File.separator + connectorName + ".json")).parallel() // for
```

Comment:
I think that accessing this way doesn't work with a bundled jar. It is working only in development mode with an exploded folder. Next to use some getResourceAsStream methods

I will have a look to provide PR

Reply:
We can also use the catalog helper.

Reply:
From core catalog.